config-provisioning/src/main/java/com/yahoo/config/provision/ApplicationId.java

```diff
@@ -54,18 +59,23 @@ public static ApplicationId from(String tenant, String application, String insta
     }
 
     public static ApplicationId fromSerializedForm(String idString) {
-        String[] parts = idString.split(":");
-        if (parts.length < 3)
-            throw new IllegalArgumentException("Application ids must be on the form tenant:application:instance, but was " + idString);
-
-        return from(parts[0], parts[1], parts[2]);
+        return fromIdString(idString, ":");
     }
 
     public static ApplicationId fromFullString(String idString) {
-        String[] parts = idString.split("\\.");
-        if (parts.length < 3)
-            throw new IllegalArgumentException("Application ids must be on the form tenant.application.instance, but was " + idString);
+        return fromIdString(idString, "\\.");
+    }
 
+    private static ApplicationId fromIdString(String idString, String splitCharacter) {
+        String[] parts = idString.split(splitCharacter);
+        String errorMessage = "Application ids must be on the form tenant" +
+                splitCharacter + "application" + splitCharacter + "instance, but was " + idString;
```

Comment:
Note that this will print a slightly confusing application ID: "... the form tenant\\.application\\.instance, but..."

Reply:
Yes, good point, fixed and made sure to actually test both forms